cloudinary-image-management-and-manipulation-in-the-cloud-cdn/ui-definitions/tabs/dashboard-content.php

```diff
@@ -7,6 +7,27 @@
 
 $video_url = ''; // Left blank for when we get the final video URL.
 $usage     = $this->plugin->components['connect']->usage;
+// Different account version have usage limits.
+if ( isset( $usage['credits'] ) ) {
+	// New credit system.
+	$usage['credits']         = array(
+		'limit'        => $usage['credits']['limit'],
```

Comment:
@DavidCramer I notice that we are checking if the `credits` key is set before we run this logic, but I wonder if that's enough. My concerns here are what about all these child keys, that might not be set. To add to it, it seems that we are getting this information from a source that we don't control. Ideally, this should be addressed on `usage_stats()` — setting some defaults —, so we can be relaxed using the logic elsewhere. Thoughts?

Reply:
@pereirinha - That is a good suggestion. However, the issue is that there are two different style of settings, although we could switch between the defaults.
Also, this is directly from their API, and is documented. So it shouldn't really change. If it did, we would have it break without breaking.

An ideal way, would be to check when calling the API, then output an error before actually setting the value.
I'll let this pass for now, then loop back with a clean up ticket.